microprofile/cdi/src/main/java/io/helidon/microprofile/cdi/BuildTimeInitializer.java

```diff
@@ -25,6 +28,8 @@
 final class BuildTimeInitializer {
     private static volatile HelidonContainerImpl container;
 
+    private static final ReentrantReadWriteLock CONTAINER_ACCESS = new ReentrantReadWriteLock();
```

Comment:
Reviewing as I go, of course; I wonder if you'll need a read and a write lock.  If not, consider a simple `Lock` instead.

Reply:
You're right. A `ReentrantLock` is sufficient.